pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -35,6 +36,7 @@ const (
 	NodeSkipped   NodePhase = "Skipped"
 	NodeFailed    NodePhase = "Failed"
 	NodeError     NodePhase = "Error"
+	NodeMemoized  NodePhase = "Memoized"
```

Comment:
Do we really need `memoized` status?  I discussed with @jessesuen. He felt adding a new state in the state machine is too complicated. Can you provide us the benefit of this new status?

Reply:
I had to distinguish memoized steps to skip node existence check when I started to implement this feature. Since I changed the logic after your first review comment, it may not be necessary. I'll try to remove it out.

Reply:
https://github.com/argoproj/argo/blob/master/workflow/controller/operator.go#L606

If we want to use stored templates for failed pods without another phase, the pods can be considered as `pod deleted` if they have been deleted. I'll try to exclude memoized case without adding another phase.

Reply:
I will use empty `StartedAt` for memoized nodes.